irods/test/data_obj_test.py

```diff
@@ -1517,6 +1517,33 @@ def enter_file_into_irods( session, filename, **kw_opt ):
                 self.sess.resources.remove( resc_name )
         self.assertIs( default_XML_parser(), current_XML_parser() )
 
+    def test_access_through_resc_hierarchy_243(self):
+        s = self.sess
+        data_path = '{coll}/{data}'.format(coll = helpers.home_collection(s),
+                                           data = unique_name(my_function_name(),datetime.now()))
+        try:
+            parent = s.resources.create('parent','deferred')
+            with self.create_simple_resc('resc0_243') as r0, \
+                 self.create_simple_resc('resc1_243') as r1:
+                s.resources.add_child(parent.name,r0)
+                s.resources.add_child(parent.name,r1)
+                hiers = ['parent;{0}'.format(r) for r in (r0,r1)]
+                # write two different replicas
+                for hier in hiers:
+                    opts = {kw.RESC_HIER_STR_KW: hier}
+                    with s.data_objects.open(data_path, 'a', **opts) as obj_io:
+                        obj_io.seek(0)
+                        obj_io.write(hier.encode('utf-8'))   # Write different content to each replica
```

Comment:
note in a comment also that the first replica will end up 'stale' and the second replica will be 'good'.